src/acl.c

```diff
@@ -640,34 +746,75 @@ sds ACLDescribeUserCommandRules(user *u) {
      * predicted bitmap is exactly the same as the user bitmap, and abort
      * otherwise, because aborting is better than a security risk in this
      * code path. */
-    if (memcmp(fakeuser->allowed_commands,
-                        u->allowed_commands,
-                        sizeof(u->allowed_commands)) != 0)
+    if (memcmp(fake_selector->allowed_commands,
+                        selector->allowed_commands,
+                        sizeof(selector->allowed_commands)) != 0)
     {
         serverLog(LL_WARNING,
             "CRITICAL ERROR: User ACLs don't match final bitmap: '%s'",
             rules);
-        serverPanic("No bitmap match in ACLDescribeUserCommandRules()");
+        serverPanic("No bitmap match in ACLDescribeSelectorCommandRules()");
     }
     return rules;
 }
 
-/* This is similar to ACLDescribeUserCommandRules(), however instead of
+sds ACLDescribeSelector(aclSelector *selector) {
+    listIter li;
+    listNode *ln;
+    sds res = sdsempty();
+    /* Key patterns. */
+    if (selector->flags & SELECTOR_FLAG_ALLKEYS) {
+        res = sdscatlen(res,"~* ",3);
+    } else {
+        listRewind(selector->patterns,&li);
+        while((ln = listNext(&li))) {
+            keyPattern *thispat = (keyPattern *)listNodeValue(ln);
+            if (thispat->flags == (CMD_KEY_READ | CMD_KEY_WRITE)) {
+                res = sdscatlen(res,"~",1);
+            } else if (thispat->flags == CMD_KEY_READ) {
+                res = sdscatlen(res,"%R~",3);
+            } else if (thispat->flags == CMD_KEY_WRITE) {
+                res = sdscatlen(res,"%W~",3);
```

Comment:
As per your initial design, there was a %D  to denote `delete`.

Reply:
For now I'm thinking about dropping %D. The motivation is that it's hard to document exactly what the difference between write and delete are. R/W also cover most of the interesting cases.